api/src/middleware/authorization.js

```diff
@@ -23,6 +23,7 @@ module.exports = {
       .getUserCtx(req)
       .then(userCtx => {
         req.userCtx = userCtx;
+        req.replicationId = req.headers['medic-replication-id'];
```

Comment:
This is only used in one place as far as I can see - instead of setting it in the middleware can we just access the header in the controller? That way it won't execute for non-purging requests.

Reply:
I had that idea some time ago about making an initial replication cache. The first step was to consistently and uniquely identify the user + device they were replicating from.
This is the hole this header wanted to fill. 

If you don't like it, since purging is calling it's own endpoints, I can easily just abandon the header and use a query parameter.